langchain4j-dashscope/src/main/java/dev/langchain4j/model/dashscope/QwenHelper.java

```diff
@@ -36,10 +41,112 @@ static List<Message> toQwenMessages(Iterable<ChatMessage> messages) {
     static Message toQwenMessage(ChatMessage message) {
         return Message.builder()
                 .role(roleFrom(message))
-                .content(message.text())
+                .content(toSingleText(message))
                 .build();
     }
 
+    static String toSingleText(ChatMessage message) {
+        switch (message.type()) {
+            case USER:
+                UserMessage userMessage = (UserMessage) message;
+                return userMessage.hasSingleText() ?
+                        ((TextContent) userMessage.contents().get(0)).text() : "";
```

Comment:
should it concatenate all text contents if there are more than 1? (instead of returning empty string)
or throw an exception if it is not allowed

Reply:
I once thought that the Qwen models would return a response to further confirm the user's intent, when it receives an empty string. However, it turns out this is not the case after actual testing. Furthermore, I've noticed that langchain4j's TextContent also does not allow empty strings. So an exception should indeed be thrown here.

Concatenating all the text (using '\n') also make sense. BTW, I recommend that UserMessage retains the text() method, which also concatenates TextContent and returns it, so that all ChatMessages can have a consistent interface and avoid having many switch-cases in the application which depends on langchain4j.

Reply:
Yeah i think i prematurely annotated text() as @Deprecated, probably we should still keep it, at least while multimodal models are still rare. The intent was to force the user to use message-type-specific accessors, because fields thst each message type comtains diverge more and more and Inam not sure it is a good idea to keep accessing all specific fields via the same getter. Moreover, user message becomes multimodal and aimessage will probably become soon too.